components/contract-functions/contract-function.tsx

```diff
@@ -110,30 +115,35 @@ export const ContractFunction: React.FC<ContractFunctionProps> = ({
   );
 };
 
-interface ContractFunctionsPanelProps {
-  functions: AbiFunction[];
-  contract?: SmartContract;
-}
+type ContractFunctionsPanelProps =
+  | { functions: AbiFunction[]; events?: undefined; contract?: SmartContract }
+  | { functions?: undefined; events: AbiEvent[]; contract?: SmartContract };
```

Comment:
why can't we just use the same input? you're merging them below anyways

you can rename it from "functions" to "input" or "fnsOrEvents" or anythigne lse really